Pushing changes made to Get Started flow and guides (#229)
atsign-foundation/atsign.dev: content/en/docs/guides/dess-setup/dess-aws/_index.md

---
title: "Setup dess on AWS (Amazon Web Services) Platform"
SEOtitle: "Setup @platform (at_platform or AtPlatform) dess in AWS (Amazon Web Services)"
linkTitle: "AWS"
weight: 2
date: 2021-07-26
content: The @platform resources will help you quickly build your end-to-end encrypted app
description: Step-by-Step setup of dess on AWS (Amazon Web Services)
---

# Step-by-Step setup of dess in AWS

> In this step-by-step guide I will walk you through all steps required to setup your own private dess in AWS cloud from scratch. Please use index to skip some steps in case you have done them in other way.

Reviewer: ```suggestion
In this step-by-step guide I will walk you through all steps required to setup your own private dess on AWS. Please use index to skip some steps in case you have done them another way.
```

> ### 2. Sign-up for GCP account <a name="sign_up"></a>

Reviewer: ```suggestion
### 2. Sign-up for AWS account <a name="sign_up"></a>
```

> If you are new to cloud like me and need to create new AWS account, I have good news! The creation is for free and the cost of running your @sing dess costs about 10$/month. You can create your free account at https://aws.amazon.com/ at date of writing 6-Jun-2021 there is “free tier” available and was used for this setup.

Reviewer: ```suggestion
If you are new to cloud like me and need to create new AWS account, I have good news! Account creation is free and running a dess instance costs about 10$/month. You can create your account at https://aws.amazon.com/ at the date of writing (6-Jun-2021) there is a “free tier” available and it was used for this setup.
```

![image-20210726083635919](images/image-20210726083635919.png)

> Based on the name you selected and the domain AWS will give you options of free domains and their pricing. As I am looking for best deal .link domain seems like good option at cost of 5$ / year.

Reviewer: ```suggestion
Based on the domain name you selected, AWS will give you other options. As I am looking for the best deal, the .link domain seems like a good option at a cost of 5$ / year.
```

This can take some time so why don’t we move on to next step!

### 4. Preparing AWS instance <a name="prep_instance"></a>

> Now since I am new to AWS the easiest way to start is using the LightSail service from service catalog. This will enable you to deploy small system which is more then capable of handling dess at pre-set price.

Reviewer: ```suggestion
Now since I am new to AWS, the easiest way to start is by using the LightSail service from their service catalog. This will enable you to deploy a small system which is more than capable of handling dess at a pre-set price.
```

![img](images/clip_image006-16272842283483.jpg)

> Next up will be selection of operating system we want to deploy. We know that dess works well with Ubuntu 20.04 LTS so lets select just that.

Reviewer: ```suggestion
Next up will be selection of operating system we want to deploy. We know that dess works well with Ubuntu 20.04 LTS so let's select just that.
```

![img](images/clip_image008.jpg)

You are presented with couple more options, but unless you know what you are doing leave these as is.

![img](images/clip_image010.jpg)

> Now let’s select instance plan. dess is relatively light weight so for testing purposes I will select first instance plan for 3.5$/Month. This will provide us with 512 MB of RAM, 1vCPU, 20GB of storage and 1TB of data transfer. This is more then enough to run our dess.

Reviewer: ```suggestion
Now let’s select an instance plan. dess is light weight, so for testing purposes I will select first instance plan for 3.5$/Month. This will provide us with 512 MB of RAM, 1vCPU, 20GB of storage and 1TB of data transfer. This is more then enough to run dess.
```

![img](images/clip_image014.png)

> At this point we have created DNS record we will use to link our dess, we created instance name which will be running our dess and we have opened port range which is exposed to the internet and we can communicate with @sign root server and our apps with.

Reviewer: ```suggestion
At this point we have created the DNS record that we will use to link dess, we created an instance name which will be running our dess and we have opened the port range which is exposed to the internet and we can communicate with the @sign root server, and our apps.
```

#### c) Setting up Firewall <a name="firewall"></a>

> Next up we need to make sure we have ports open for our dess to communicate with root server and our apps. In Section networking go to section “IPv4 Firewall” and click “+ Add rule” Our rule will be “Custom” on TCP protocol with Port range in number higher then 1024. In my case I have selected port range 8000-8010. This will enable me to run up to 10 @signs in parallel.

Reviewer: ```suggestion
Next up, we need to make sure we have ports open for dess to communicate with the root server and our apps. In the networking section go to “IPv4 Firewall” and click “+ Add rule”. The rule will be “Custom” on TCP protocol with a Port range starting with a number higher than 1024. In my case I have selected port range 8000-8010. This will enable me to run up to 11 @signs in parallel.
```

To make it more understandable:

> I will be registering my @sing **@44likelycanary**

Reviewer: ```suggestion
I will be registering my @sign **@44likelycanary**
```
